### ISSUE TITLE
Miscalculated carousel width in tables

### ISSUE THREAD

**Issue by Christoph-Wagner** [fancyapps/ui#544]
### Describe the bug

Demo: https://codepen.io/cwagner/pen/qBQvbXP

Inside a table, the carousel calculates the width wrong, resulting in a far too wide table.
The example also shows that it’s somehow related to the amount or combined width of all the images in the slider.
This is fixed by giving the `.f-carousel` container a fixed width, which actually solves the issue for us, for now, but won’t when we use the carousel elsewhere.

### Reproduction

1. Create a container div with a certain width like 600px
2. Place a table inside it with no extra styling. 
3. Now place a carousel inside a table cell and initialize it. The Carousel becomes far wider than the table

### Additional context

![image](https://github.com/fancyapps/ui/assets/1609329/56849f6f-188f-44b1-a433-7578ed1bd7db)

**Comment by fancyapps** [comment]
The script does not define any width, you do it yourself, otherwise it will freely occupy the available space, because it is just how CSS works.

**Comment by Christoph-Wagner** [comment]
But there is a width, the 600px from the container. The issue is that it takes *more* than 100% of the width, and that way making the table broader. If you put an image inside the table with 800px width, and give it `max-width: 100%`, it becomes 600px wide, because that’s how much space there is. There is no such thing for the carousel.

**Comment by fancyapps** [comment]
I will repeat - you control the width using your CSS.

**Comment by Christoph-Wagner** [comment]
Yes, the css has `width:670px` in the container. The carousel ignores that. That’s what I’m saying.

**Comment by Christoph-Wagner** [comment]
Not for the table, no, for the parent of that table. I know tables behave differently, but it is possible to limit elements automatically to the width of the normal table (and in most cases this does work automatically, `white-space: nowrap` is one of the ways this breaks, just like a lack of breakable white space in general), but I guess the carousel does not support that.

> Basically, give some width for your <td>

This does not work. I extended the demo to show that setting the width of the `td` element changes nothing, nor does setting the width of the `table` element. The only solution here would be to set the whole table to `table-layout:fixed` which sadly comes with a lot of other issues.
